packages/edit-site/src/components/header/index.js

```diff
@@ -79,64 +84,110 @@ export default function Header( {
 	const displayBlockToolbar =
 		! isLargeViewport || deviceType !== 'Desktop' || hasFixedToolbar;
 
+	// Track mouse location.
+	const [ mouseLoc, setMouseLoc ] = useState( { x: 0, y: 0 } );
+	useEffect( () => {
+		const setMouseLocFromEvent = ( event ) =>
+			setMouseLoc( { x: event.clientX, y: event.clientY } );
+
+		window.addEventListener( 'mousemove', setMouseLocFromEvent );
+
+		return () => {
+			window.removeEventListener( 'mousemove', setMouseLocFromEvent );
+		};
+	}, [] );
+
+	// See if a template part is hovered when mouse moves.
+	const hoveredTemplatePartRef = useRef( '' );
+	useEffect( () => {
```

Comment:
We could probably get rid of this ref and just `useMemo` instead of `useEffect`.  Or even just declare a variable if we don't think it is computationally expensive enough to warrant memoization with mouse location dependencies? 🤔